### ISSUE TITLE
Error: Could not resolve encryptionManager!  with PicoCMS and Nextcloud >= 20 

### ISSUE THREAD

**Issue by phaus** [nextcloud/cms_pico#153]
After upgrading to Nextcloud 20, the log is flodded with errors like:

> OCP\AppFramework\QueryException: Could not resolve encryptionManager! Class encryptionManager does not exist

Disabling the PicoCMS seems to resolve the issue. 
I will try to dig into it, to find more about the root cause.

There is already an issue in the Nextcloud forum: https://help.nextcloud.com/t/fehler-could-not-resolve-encryptionmanager/100439

**Comment by phaus** [comment]
I got rid of that error by updating the file `cms_pico/lib/Listener/ExternalStorageBackendEventListener.php`

I did change:

`use OC\Encryption\Manager as EncryptionManager;` with `use OCP\Encryption\IManager as EncryptionManager;`

But there seems to be another issue with the Bootstraping, since [AppInfo is deprecated for Version >= 20](https://docs.nextcloud.com/server/latest/developer_manual/app_development/bootstrap.html#nextcloud-20-and-later):

```json
{
    "reqId": "ewrwerwerwer",
    "level": 0,
    "time": "2020-12-08T00:17:11+01:00",
    "remoteAddr": "XXXXXX",
    "user": "user",
    "app": "cms_pico",
    "method": "POST",
    "url": "/apps/text/session/sync",
    "message": "/appinfo/app.php is deprecated, use \\OCP\\AppFramework\\Bootstrap\\IBootstrap on the application class instead.",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.16; rv:83.0) Gecko/20100101 Firefox/83.0",
    "version": "20.0.3.0"
}
```

**Comment by lvarnava** [comment]
I have the same problem.
NC 19.0.5
Ubuntu 18.04
Postgresql 10.15
Nginx 1.14
PHP 7.2.24
CMS_PICO 1.0.11

[no app in context] Error: OCP\AppFramework\QueryException: Could not resolve encryptionManager! Class encryptionManager does not exist at <<closure>>

 0. /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php line 116
    OC\AppFramework\Utility\SimpleContainer->resolve("encryptionManager")
 1. /var/www/nextcloud/lib/private/ServerContainer.php line 149
    OC\AppFramework\Utility\SimpleContainer->query("encryptionManager", true)
 2. /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php line 78
    OC\ServerContainer->query("encryptionManager")
 3. /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php line 101
    OC\AppFramework\Utility\SimpleContainer->buildClass(ReflectionClass  ... "})
 4. /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php line 116
    OC\AppFramework\Utility\SimpleContainer->resolve("OCA\\CMSPico\\L ... r")
 5. /var/www/nextcloud/lib/private/ServerContainer.php line 149
    OC\AppFramework\Utility\SimpleContainer->query("OCA\\CMSPico\\L ... r", true)
 6. /var/www/nextcloud/lib/private/EventDispatcher/ServiceEventListener.php line 66
    OC\ServerContainer->query("OCA\\CMSPico\\L ... r")
 7. /var/www/nextcloud/3rdparty/symfony/event-dispatcher/EventDispatcher.php line 251
    OC\EventDispatcher\ServiceEventListener->__invoke(OCP\EventDispatcher\GenericEvent {}, "OCA\\Files_Exte ... s", Symfony\Componen ... {})
 8. /var/www/nextcloud/3rdparty/symfony/event-dispatcher/EventDispatcher.php line 73
    Symfony\Component\EventDispatcher\EventDispatcher->callListeners([Closure {},Closure {}], "OCA\\Files_Exte ... s", OCP\EventDispatcher\GenericEvent {})
 9. /var/www/nextcloud/lib/private/EventDispatcher/EventDispatcher.php line 86
    Symfony\Component\EventDispatcher\EventDispatcher->dispatch(OCP\EventDispatcher\GenericEvent {}, "OCA\\Files_Exte ... s")
10. /var/www/nextcloud/lib/private/EventDispatcher/SymfonyAdapter.php line 63
    OC\EventDispatcher\EventDispatcher->dispatch("OCA\\Files_Exte ... s", OCP\EventDispatcher\GenericEvent {})
11. /var/www/nextcloud/apps/files_external/lib/Service/BackendService.php line 118
    OC\EventDispatcher\SymfonyAdapter->dispatch("OCA\\Files_Exte ... s", OCP\EventDispatcher\GenericEvent {})
12. /var/www/nextcloud/apps/files_external/lib/Service/BackendService.php line 125
    OCA\Files_External\Service\BackendService->callForRegistrations()
13. /var/www/nextcloud/apps/files_external/lib/Service/BackendService.php line 230
    OCA\Files_External\Service\BackendService->loadBackendProviders()
14. /var/www/nextcloud/apps/files_external/lib/Service/StoragesService.php line 304
    OCA\Files_External\Service\BackendService->getBackend("sftp")
15. /var/www/nextcloud/apps/files_external/lib/Service/StoragesService.php line 104
    OCA\Files_External\Service\StoragesService->createStorage("/mdalib", "sftp", "password::password", {password: "Ny8U ... "}, {encrypt: true,p ... e}, [], ["mdausers"], 100)
16. <<closure>>
    OCA\Files_External\Service\StoragesService->getStorageConfigFromDBMount({mount_id: 19,mo ... }})
17. /var/www/nextcloud/apps/files_external/lib/Service/UserGlobalStoragesService.php line 188
    array_map([OCA\Files_Exter ... "], [{mount_id: 19,m ... }])
18. /var/www/nextcloud/apps/files_external/lib/Config/ConfigAdapter.php line 129
    OCA\Files_External\Service\UserGlobalStoragesService->getAllStoragesForUser()
19. /var/www/nextcloud/lib/private/Files/Config/MountProviderCollection.php line 105
    OCA\Files_External\Config\ConfigAdapter->getMountsForUser(OC\User\User {}, OC\Files\Storage\StorageFactory {})
20. /var/www/nextcloud/lib/private/Files/Filesystem.php line 451
    OC\Files\Config\MountProviderCollection->addMountForUser(OC\User\User {}, OC\Files\Mount\Manager {})
21. /var/www/nextcloud/lib/private/Files/Filesystem.php line 376
    OC\Files\Filesystem::initMountPoints("lvarnava")
22. /var/www/nextcloud/lib/private/legacy/OC_Util.php line 308
    OC\Files\Filesystem::init("****", "/***/files")
23. /var/www/nextcloud/lib/base.php line 1009
    OC_Util::setupFS()
24. /var/www/nextcloud/index.php line 37
    OC::handleRequest()

**Comment by phaus** [comment]
What about the Bootstraping thing?
Will it be handled in another issue?
The Plugin does not work in Nextcloud 20 atm.

**Comment by PhrozenByte** [comment]
Fixed in [`v1.0.12`](https://github.com/nextcloud/cms_pico/releases/tag/v1.0.12). Thank you for reporting this! :+1:

**Comment by PhrozenByte** [comment]
> What about the Bootstraping thing?
> Will it be handled in another issue?
> The Plugin does not work in Nextcloud 20 atm.

You can (and should) open another issue for this if you want to, but it will be marked as `status: Defer` for now. It's no bug, this is just a new feature of Nextcloud we can't use until Nextcloud 20 is the lowest still supported version. You're seeing this in your logfiles because you've debug mode enabled. Disable debug mode.

It *does* work with Nextcloud 20 just fine.
